#!/usr/bin/env python3
PACKAGE = "orb_slam2_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enable_localization_mode", bool_t, 0, "Enable localization mode, Disable mapping", True)
gen.add("save_trajectory_on_exit", bool_t, 0, "Save trajectory when SLAM system has shut down", False)
gen.add("scale_factor", double_t, 0, "Scale factor for SLAM estimates", 1.0, 0.1, 5.0)
gen.add("min_observations_per_point", int_t, 0, "Minimum number of observations per point", 2, 1, 10)

exit(gen.generate(PACKAGE, "orb_slam2_ros", "dynamic_reconfigure"))